components/python/pygobject-3/pygobject-3-PYVER.p5m

```diff
@@ -11,47 +11,75 @@
 
 #
 # Copyright 2016 Alexander Pyhalov
+# Copyright 2020 Andreas Wacknitz
 #
 
 set name=pkg.fmri value=pkg:/library/python/pygobject-3-$(PYV)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
 set name=pkg.summary value="Python bindings for GObject"
 set name=info.classification value="org.opensolaris.category.2008:Development/Python"
-set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
 set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
 set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 
 license COPYING license="LGPL v2.1"
 
 depend type=require fmri=library/python/pygobject-3@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
 depend type=require fmri=library/desktop/gtk3@3.14.15,5.11-2016.0.0.2
 
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/64/_gi.so 
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/64/_gi_cairo.so
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/__init__.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/_constants.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/_error.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/_gi.so
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/_gi_cairo.so
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/_gobject/__init__.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/_option.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/_propertyhelper.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/_signalhelper.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/docstring.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/importer.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/module.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/GIMarshallingTests.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/GLib.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/GObject.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/Gdk.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/Gio.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/Gtk.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/Pango.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/__init__.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/overrides/keysyms.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/pygtkcompat.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/repository/__init__.py
-file path=usr/lib/python$(PYVER)/vendor-packages/gi/types.py
-file path=usr/lib/python$(PYVER)/vendor-packages/pygobject-$(COMPONENT_VERSION)-py$(PYVER).egg-info
-file path=usr/lib/python$(PYVER)/vendor-packages/pygtkcompat/__init__.py
-file path=usr/lib/python$(PYVER)/vendor-packages/pygtkcompat/generictreemodel.py
-file path=usr/lib/python$(PYVER)/vendor-packages/pygtkcompat/pygtkcompat.py
+$(PYTHON_2.7_ONLY)file path=usr/lib/python$(PYVER)/vendor-packages/64/gi/__init__.py
```

Comment:
Does it work as expected fo python 2.7?
Note layout - only *.so files are in `usr/lib/python$(PYVER)/vendor-packages/gi/64`

Reply:
If I change the Makefile
#VARIANTS_64 = $(VARIANT_PYTHON27) $(VARIANT_PYTHON35)
VARIANTS_64 = $(VARIANT_PYTHON27)

then "gmake publish" works.

I don't understand your 2nd comment regarding only *.so files.

Reply:
*.py files for 64-bit python 2.7 should likely live in usr/lib/python$(PYVER)/vendor-packages/*, for example, `usr/lib/python2.7/vendor-packages/gi/types.py`, but for *.so files last directory in path should be `64`, for example, `usr/lib/python2.7/vendor-packages/gi/64/_gi.so` (and not `usr/lib/python2.7/vendor-packages/64/gi/types.py` or `usr/lib/python2.7/vendor-packages/64/gi/_gi.so` , where setup will likely place them).